### ISSUE TITLE
Minimal requirements check

### ISSUE THREAD

**Issue by Pryx** [Pryx/server-status#31]
We should check (ideally both before install and in administration every time) for minimal requirements - #18 will surely need CURL to work and we should check if `mail()` function exists and is working. Maybe send testing email with "Welcome" information right after install? I don't think server memory or max script run time will be problem (yet). Also maybe checking write permission before trying to write config file could be nice. If anyone has more ideas what to check, let me know.

**Comment by jhuesser** [comment]
Maybe it's anoying for a user to get all the mails. So you could test if ```mail()``` is present with sending it to a dummy address.
```php
$to      = 'nobody@example.com';
$subject = 'the subject';
$message = 'hello';
if(mail($to, $subject, $message){
   echo "mail() enabled"; //but mail will not be delivered
} else {
  echo "mail() disabled";
}
```

**Comment by Pryx** [comment]
I wanted to send email as a second layer of checking - the mail server can be misconfigured and the mail function would still return true, which would be a problem for users who forget their passwords for example :slightly_smiling_face:

**Comment by jhuesser** [comment]
Ah, yeah then it's the best to send a mail to the user.

**Comment by petrk94** [comment]
@Pryx ah ok, I could write that short script, but if you make it in PHP directly, than it make more sense^^
Just, I cant PHP, but love that this cool software come from my home country and so I follow the developments :)

**Comment by thnilsen** [comment]
Would something like this do for basic prerequsite checks during installation?

The example checks the following three points:

- PHP version (>=5.4.0) 
- If mysqlnd library is installed by checking if one of its unique functions are available.
- If the process has write access to the web folder. 

Other checks can be easily added if needed. 
Check for dbname,dbuser and dbpass validity would have to be handled differently and after the form has been posted..

I haven't commited the update yet, so the diff to install.php is: 
[install.php.diff.txt](https://github.com/Pryx/server-status/files/2554952/install.php.diff.txt)


The end result looks like this
![prereq](https://user-images.githubusercontent.com/10619272/48091669-9b6d9300-e20a-11e8-9105-84ad81155a9d.png)

This should fix #35 as well.
